W-14568574 salesforce updates (#2444)
mulesoft/docs-connectors: salesforce/0.3.9/modules/ROOT/pages/index.adoc

=== Message Durability

Salesforce stores high-volume platform events and change events for 72 hours, and standard-volume events (the legacy PushTopic and generic events) for 24 hours, so you can retrieve stored events during that retention window. The Streaming API event framework decouples event producers from event consumers.
A subscriber can retrieve events at any time and isn’t restricted to listening to events at the time they’re sent.

> Salesforce assigns a replay ID to each broadcast event, which you can use to replay stored event messages. For more information, refer to https://developer.salesforce.com/docs/atlas.en-us.api_streaming.meta/api_streaming/using_streaming_api_durability.htm[Message Durability] in the Streaming API Developer Guide.

Reviewer: ```suggestion
Salesforce assigns a replay ID to each broadcast event, which you can use to replay stored event messages. For more information, refer to https://developer.salesforce.com/docs/atlas.en-us.api_streaming.meta/api_streaming/using_streaming_api_durability.htm[Message Durability^] in the Streaming API Developer Guide.
```

=== Replay Events from a Topic

> A subscriber can choose which events to receive, such as all events within the retention window or starting after a particular event. The default is to receive only the new events sent after subscribing. Events outside the retention period are discarded.

Reviewer 2: ```suggestion
A subscriber can choose which events to receive, such as all events within the retention window or events that start after a particular event. The default is to receive only the new events sent after subscribing. Events outside of the retention period are discarded.
```

Replay options:

. Replay ID	Subscriber receives all events after the event specified by its `replayId` value.

> . -1	- Subscriber receives new events that are broadcast after the client subscribes.

Reviewer 2: What do -1 and -2 mean? This is a very confusing presentation. Are the numbers part of the choices, or should these be regular bullets? This comment also applies to existing text.

Author: These are values. This is also a Mule 3 doc. I fixed the formatting though

The connector also supports automatic replay of stored events, based on the replay id of the last event that has been processed by the connector. This has proved useful in cases when the connector stopped listening for some reason (server shutdown, connection dropped).

By having the flag "Resume from the Last Replay Id" set to true, upon starting, the connector will replay all the events starting with the last processed event's replay id.

> If the stored replay ID is outside this event retention window, then the replay option selected by the user will determine what events will be replayed.

Reviewer 2: ```suggestion
If the stored replay ID is outside this event retention window, then the replay option selected by the user determines what events will be replayed.
```
